packages/astro/src/@types/astro.ts

```diff
@@ -1555,10 +1555,19 @@ export interface AstroUserConfig {
  */
 export type InjectedScriptStage = 'before-hydration' | 'head-inline' | 'page' | 'page-ssr';
 
+/**
+ * IDs for different priorities of injected routes and redirects:
+ * - "above-project": Override any project route in case of conflict.
+ * - "same-as-project": Behave the same as if the route was defined in the project, following the same priority rules.
+ * - "below-project": Only match if no project route matches.
```

Comment:
@withastro/maintainers-docs You should focus here. Naming is very important here

Reply:
I think the naming should involve file-based routing somehow, rather than project. Took me a second to realize that's what project meant.

Reply:
I also wonder if “above”/“below” is ideal. I understand why it was chosen, but I associate high/low more with priority.

Maybe something like this could work? @sarah11918 is also a good person to bounce these off. I don’t _love_ “file-system” here — not sure it’s super clear that this means “file-based route” or if there’s something clearer.

Could ever be other kinds of priority? If not, it could be redundant to mention that in the enum values and the options could just be `high`/`default`/`low` or something? With the relationship explained in the documentation.

```suggestion
 * IDs for different priorities of injected routes and redirects:
 * - "higher-than-file-system": Override any project route in case of conflict.
 * - "same-as-file-system": Behave the same as if the route was defined in the project, following the same priority rules.
 * - "lower-than-file-system": Only match if no project route matches.
```

Reply:
Another option would be to use a number, with all project routes having priority zero, injected routes defaulting to a priority of -1 and redirects defaulting to a priority of 1.

Then the other rules apply for when this number is the same.

This may be easier to document and even more flexible. Maybe too flexible and would make things confusing for integration developers.

Reply:
Yes, I’d caution against that I think. Even if internally the enum maps to `-1 | 0 | 1`, allowing people to potentially try to outbid each other with ever higher numbers seems really risky. (And doesn’t actually help with clarity `high` or `higher-than-file-system` tell you something quite explicit, where `1` is pretty meaningless on its own.)

Reply:
Agreed, I also don't like the numbers idea.

Reply:
@Fryuni OK, so my gut reaction was to look at your definitions to see what terms might make sense. The first thing that comes to mind is something like:

Injected/Redirect Route Priorities:
- `override` - takes higher priority than other defined routes in your project
- `default` - integrates/conforms with the other routes defined in your project
- `defer` - route is only prioritized after all other normally-defined routes in your project

(Admittedly, I had the hardest time thinking of the last one) Maybe that gives you another direction to think in?

Reply:
I like that direction @sarah11918!

Reply:
I also liked @sarah11918's suggestion. I'll send a commit with those and change from just "project routes" to "project file-based routes" to make it clear what they are relative to.

Reply:
Just sent a commit with a new suggestion. I first used the terms by @sarah11918, but it felt weird to have a priority named "default" that is not the default, so I went with "merge".

Also changed the "project routes" to clarify that those are the routes inferred from files.

What do you all think? Better, worse, still confusing?

Reply:
Right, I agree not to use default as an option that is not the default.  😀 

`merge` could work. Synonyms would be like `integrate` or `incorporate` or maybe even `include`.

If you wanted the "conform" (follow the rules) angle, then other synonyms could be `follow`, `obey`, `observe`, `match`, `comply`.

I'm not sure I have a strong preference. Just highlighting some options to see if one immediately jumps out at you!

Reply:
Out of those, I think I still prefer `merge`, with `incorporate` as a close second.